catalog/README.md

```diff
@@ -0,0 +1,159 @@
+# Model Catalog Service
+
+The Model Catalog Service provides a **read-only discovery service** for ML models across multiple catalog sources. It acts as a federated metadata aggregation layer, allowing users to search and discover models from various external catalogs through a unified REST API.
+
+## Architecture Overview
+
+The catalog service operates as a **metadata aggregation layer** that:
+- Federates model discovery across different external catalogs
+- Provides a unified REST API for model search and discovery
+- Uses pluggable source providers for extensibility
+- Operates without traditional database storage (file-based configuration)
+
+### Supported Catalog Sources
+
+- **YAML Catalog** - Static YAML files containing model metadata
+- **Red Hat Ecosystem Catalog (RHEC)** - GraphQL API integration for container and model discovery
```

Comment:
maybe highlight this is considered a "example"/reference implementation of how people can extend with their own graphql providers if needed by following the remainder of this guide?